amqp_job_queue.go

```diff
@@ -84,7 +84,8 @@ func (q *AMQPJobQueue) Jobs(ctx gocontext.Context) (outChan <-chan Job, err erro
 			select {
 			case <-ctx.Done():
 				return
-			case delivery, ok := <-deliveries:
+			case <-ready:
+				delivery, ok := <-deliveries
```

Comment:
TODO: I think that this recv from `<-deliveries` should also have a `select` timeout.